pkg/core/request/http.go

```diff
@@ -0,0 +1,94 @@
+package request
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+
+	"github.com/ksysoev/wasabi"
+)
+
+type HTTPReq struct {
+	ctx     context.Context
+	headers map[string][]string
+	url     string
+	method  string
+	body    []byte
+	id      int64
+}
+
+// NewHTTPReq creates a new HTTPReq instance with the specified method, URL, and body.
+// It takes three parameters: ctx of type context.Context, method of type string, and url of type string.
+// It also takes body of type []byte which represents the request payload.
+// It returns a pointer to an HTTPReq struct initialized with the provided method, URL, and body.
+func NewHTTPReq(ctx context.Context, method, url string, body []byte, reqID int64) *HTTPReq {
+	return &HTTPReq{
+		ctx:     ctx,
+		url:     url,
+		method:  method,
+		body:    body,
+		headers: make(map[string][]string),
+		id:      reqID,
+	}
+}
+
+func (r *HTTPReq) ID() int64 {
+	return r.id
+}
+
+// AddHeader adds a header key-value pair to the HTTP request.
+// It takes two parameters: key of type string and value of type string.
+// It appends the value to the list of values associated with the key in the headers map.
+func (r *HTTPReq) AddHeader(key, value string) {
+	r.headers[key] = append(r.headers[key], value)
+}
+
+// ToHTTPRequest converts an HTTPReq struct to an *http.Request.
+// It takes no parameters and uses the fields of the HTTPReq struct.
+// It returns a pointer to an http.Request and an error.
+// It returns an error if the http.NewRequestWithContext call fails, such as when the method or URL is invalid.
+func (r *HTTPReq) ToHTTPRequest() (*http.Request, error) {
+	bodyReader := bytes.NewReader(r.body)
+
+	req, err := http.NewRequestWithContext(r.ctx, r.method, r.url, bodyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	for key, values := range r.headers {
+		for _, value := range values {
+			req.Header.Add(key, value)
+		}
+	}
+
+	return req, nil
+}
+
+// Context returns the context associated with the HTTPReq.
+// It takes no parameters.
+// It returns a context.Context which is the context stored in the HTTPReq.
+func (r *HTTPReq) Context() context.Context {
+	return r.ctx
+}
+
+// RoutingKey constructs and returns a routing key string for the HTTP request.
+// It concatenates the HTTP method and URL of the request, separated by a space.
+// It does not return an error.
+func (r *HTTPReq) RoutingKey() string {
+	return r.method + " " + r.url
```

Comment:
```suggestion
	return r.url + " " + r.method
```
This way routing keys of same URLs share same prefix

Reply:
For now this method here only for compatibility with wasabi.Request interface, and not really in use, and I'm not sure that it'll be used, and most importantly don't know how it'll be used.


both orders have benefits depending on use case, for example if we'll make caching middleware we probably more interested in method rather than path to decide if we want to cache or not. Lets keep it as is for now, and change if we'll need other order in real use case.